core/templates/pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component.html

```diff
@@ -1,45 +1,139 @@
+        {{ opportunity?.heading | wrapTextWithEllipsis : opportunityHeadingTruncationLength }}
       </span>
-      <lazy-loading ng-show="$ctrl.opportunityDataIsLoading"></lazy-loading>
+      <lazy-loading [hidden]="!opportunityDataIsLoading"></lazy-loading>
     </div>
   </div>
-  <div class="oppia-opportunities-progress oppia-animate-fadein" ng-if="$ctrl.progressBarRequired">
+  <div class="oppia-opportunities-progress oppia-animate-fadein" *ngIf="progressBarRequired">
     <div class="oppia-opportunities-progress-bar">
-      <div ng-show="!$ctrl.opportunityDataIsLoading" class="progress">
-        <div class="progress-bar bg-success" role="progressbar" ng-style="$ctrl.progressBarStyle"></div>
+      <div [hidden]="opportunityDataIsLoading" class="progress">
+        <div class="progress-bar bg-success" role="progressbar" [ngStyle]="progressBarStyle"></div>
       </div>
-      <lazy-loading ng-show="$ctrl.opportunityDataIsLoading"></lazy-loading>
+      <lazy-loading [hidden]="!opportunityDataIsLoading"></lazy-loading>
     </div>
     <div class="oppia-opportunities-progress-label">
       <span class="protractor-test-opportunity-list-item-progress-percentage"
-            ng-show="!$ctrl.opportunityDataIsLoading">
-        (<[$ctrl.progressPercentage]>)
+            [hidden]="opportunityDataIsLoading">
+        ({{ progressPercentage }})
       </span>
-      <lazy-loading ng-show="$ctrl.opportunityDataIsLoading"></lazy-loading>
+      <lazy-loading [hidden]="!opportunityDataIsLoading"></lazy-loading>
     </div>
   </div>
-  <div class="oppia-opportunities-label-container oppia-animate-fadein"ng-if="$ctrl.labelRequired">
-    <div class="oppia-opportunities-label" ng-style="$ctrl.labelStyle" >
-      <span class="protractor-test-opportunity-list-item-label" ng-show="!$ctrl.opportunityDataIsLoading"><[$ctrl.labelText]></span>
-      <lazy-loading ng-show="$ctrl.opportunityDataIsLoading"></lazy-loading>
+  <div class="oppia-opportunities-label-container oppia-animate-fadein" *ngIf="labelRequired">
+    <div class="oppia-opportunities-label" [ngStyle]="labelStyle" >
+      <span class="protractor-test-opportunity-list-item-label" [hidden]="!opportunityDataIsLoading">{{ labelText }}</span>
+      <lazy-loading [hidden]="!opportunityDataIsLoading"></lazy-loading>
     </div>
   </div>
   <div class="oppia-opportunities-button-container oppia-animate-fadein">
     <button class="oppia-opportunities-button protractor-test-opportunity-list-item-button"
-            ng-show="!$ctrl.opportunityDataIsLoading"
-            ng-click="$ctrl.onClickActionButton($ctrl.opportunity.id)">
-      <[$ctrl.opportunity.actionButtonTitle]>
+            [hidden]="opportunityDataIsLoading"
+            (click)="clickActionButton(opportunity.id)">
+      {{ opportunity?.actionButtonTitle }}
     </button>
   </div>
 </div>
+<style>
+  .oppia-opportunities-button {
+  background-color: #ffffff;
```

Comment:
Indent this line and everything below (except the closing style tag) by two more spaces.